boa/cli/mambabuild.py

```diff
@@ -27,45 +27,57 @@
 
 solver_map = {}
 
+# e.g. package-1.2.3-h5487548_0
+dashed_spec_pattern = r"([^ ]+)-([^- ]+)-([^- ]+)"
+# e.g. package 1.2.8.*
+conda_build_spec_pattern = r"([^ ]+)(?:\ ([^ ]+))?(?:\ ([^ ]+))?"
+
+problem_re = re.compile(
+    rf"""
+        ^(?:\ *-\ +)+
+        (?:
+            (?:
+                package
+                \ {dashed_spec_pattern}
+                \ requires
+                \ {conda_build_spec_pattern}
+                ,\ but\ none\ of\ the\ providers\ can\ be\ installed
+            ) | (?:
+                package
+                \ {dashed_spec_pattern}
+                \ has\ constraint
+                \ .*
+                \ conflicting\ with
+                \ {dashed_spec_pattern}
+            ) | (?:
+                nothing\ provides
+                \ {conda_build_spec_pattern}
```

Comment:
The previous logic had `dashed_specs.append(words[-1])` here, i.e.,
```suggestion
                \ .*
```
. Which do we want?